File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-com-5-4/ElectricityTransmissionContingency.cpp
// ElectricityTransmissionContingency.cpp 
#include "ElectricityTransmissionContingency.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

ElectricityTransmissionContingency::ElectricityTransmissionContingency(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //contingencyNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* contingencyNode = xmlNode->FirstChildElement("contingency");

   if(contingencyNode){contingencyIsNull_ = false;}
   else{contingencyIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- contingencyNode , address : " << contingencyNode << std::endl;
   #endif
   if(contingencyNode)
   {
      if(contingencyNode->Attribute("href") || contingencyNode->Attribute("id"))
      {
          if(contingencyNode->Attribute("id"))
          {
             contingencyIDRef_ = contingencyNode->Attribute("id");
             contingency_ = boost::shared_ptr<ElectricityTransmissionContingencyType>(new ElectricityTransmissionContingencyType(contingencyNode));
             contingency_->setID(contingencyIDRef_);
             IDManager::instance().setID(contingencyIDRef_,contingency_);
          }
          else if(contingencyNode->Attribute("href")) { contingencyIDRef_ = contingencyNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { contingency_ = boost::shared_ptr<ElectricityTransmissionContingencyType>(new ElectricityTransmissionContingencyType(contingencyNode));}
   }

   //contingentPartyNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* contingentPartyNode = xmlNode->FirstChildElement("contingentParty");

   if(contingentPartyNode){contingentPartyIsNull_ = false;}
   else{contingentPartyIsNull_ = true;}

   if(contingentPartyNode)
   {
      for(contingentPartyNode; contingentPartyNode; contingentPartyNode = contingentPartyNode->NextSiblingElement("contingentParty")){
          #ifdef ConsolePrint
              FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- contingentPartyNode , address : " << contingentPartyNode << std::endl;
          #endif
          if(contingentPartyNode->Attribute("href") || contingentPartyNode->Attribute("id"))
          {
              if(contingentPartyNode->Attribute("id"))
              {
                  contingentPartyIDRef_ = contingentPartyNode->Attribute("id");
                  contingentParty_.push_back(boost::shared_ptr<PartyReference>(new PartyReference(contingentPartyNode)));
                  contingentParty_.back()->setID(contingentPartyIDRef_);
                  IDManager::instance().setID(contingentPartyIDRef_, contingentParty_.back());
              }
              else if(contingentPartyNode->Attribute("href")) { contingentPartyIDRef_ = contingentPartyNode->Attribute("href");}
              else { QL_FAIL("id or href error"); }
          }
          else { contingentParty_.push_back(boost::shared_ptr<PartyReference>(new PartyReference(contingentPartyNode)));}
      }
   }
   else {
       #ifdef ConsolePrint
           FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- contingentPartyNode , address : " << contingentPartyNode << std::endl;
       #endif
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<ElectricityTransmissionContingencyType> ElectricityTransmissionContingency::getContingency()
{
   if(!this->contingencyIsNull_){
        if(contingencyIDRef_ != ""){
             return boost::shared_static_cast<ElectricityTransmissionContingencyType>(IDManager::instance().getID(contingencyIDRef_));
        }else{
             return this->contingency_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<ElectricityTransmissionContingencyType>();
   }
}
std::vector<boost::shared_ptr<PartyReference>> ElectricityTransmissionContingency::getContingentParty()
{
   if(!this->contingentPartyIsNull_){
        if(contingentPartyIDRef_ != ""){
             return this->contingentParty_;
        }else{
             return this->contingentParty_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return std::vector<boost::shared_ptr<PartyReference>>();
   }
}
}

File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-com-5-4/ElectricityTransmissionContingency.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class ElectricityTransmissionContingency : ISerialized
    {
        public ElectricityTransmissionContingency(XmlNode xmlNode)
        {
            XmlNode contingencyNode = xmlNode.SelectSingleNode("contingency");
            
            if (contingencyNode != null)
            {
                if (contingencyNode.Attributes["href"] != null || contingencyNode.Attributes["id"] != null) 
                {
                    if (contingencyNode.Attributes["id"] != null) 
                    {
                        contingencyIDRef_ = contingencyNode.Attributes["id"].Value;
                        ElectricityTransmissionContingencyType ob = new ElectricityTransmissionContingencyType(contingencyNode);
                        IDManager.SetID(contingencyIDRef_, ob);
                    }
                    else if (contingencyNode.Attributes["href"] != null)
                    {
                        contingencyIDRef_ = contingencyNode.Attributes["href"].Value;
                    }
                    else
                    {
                        contingency_ = new ElectricityTransmissionContingencyType(contingencyNode);
                    }
                }
                else
                {
                    contingency_ = new ElectricityTransmissionContingencyType(contingencyNode);
                }
            }
            
        
            XmlNodeList contingentPartyNodeList = xmlNode.SelectNodes("contingentParty");
            
            if (contingentPartyNodeList != null)
            {
                this.contingentParty_ = new List<PartyReference>();
                foreach (XmlNode item in contingentPartyNodeList)
                {
                    if (item.Attributes["href"] != null || item.Attributes["id"] != null) 
                    {
                        if (item.Attributes["id"] != null) 
                        {
                            contingentPartyIDRef_ = item.Attributes["id"].Value;
                            contingentParty_.Add(new PartyReference(item));
                            IDManager.SetID(contingentPartyIDRef_, contingentParty_[contingentParty_.Count - 1 ]);
                        }
                        else if (item.Attributes["href"] != null)
                        {
                            contingentPartyIDRef_ = item.Attributes["href"].Value;
                        }
                        else
                        {
                        contingentParty_.Add(new PartyReference(item));
                        }
                    }
                    else
                    {
                        contingentParty_.Add(new PartyReference(item));
                    }
                }
            }
            
        
        }
        
    
        #region Contingency_
        private ElectricityTransmissionContingencyType contingency_;
        public ElectricityTransmissionContingencyType Contingency_
        {
            get
            {
                if (this.contingency_ != null)
                {
                    return this.contingency_; 
                }
                else if (this.contingencyIDRef_ != null)
                {
                    contingency_ = IDManager.getID(contingencyIDRef_) as ElectricityTransmissionContingencyType;
                    return this.contingency_; 
                }
                else
                {
                      return this.contingency_; 
                }
            }
            set
            {
                if (this.contingency_ != value)
                {
                    this.contingency_ = value;
                }
            }
        }
        #endregion
        
        public string contingencyIDRef_ { get; set; }
        #region ContingentParty_
        private List<PartyReference> contingentParty_;
        public List<PartyReference> ContingentParty_
        {
            get
            {
                if (this.contingentParty_ != null)
                {
                    return this.contingentParty_; 
                }
                else if (this.contingentPartyIDRef_ != null)
                {
                    return this.contingentParty_; 
                }
                else
                {
                      return this.contingentParty_; 
                }
            }
            set
            {
                if (this.contingentParty_ != value)
                {
                    this.contingentParty_ = value;
                }
            }
        }
        #endregion
        
        public string contingentPartyIDRef_ { get; set; }
        
    
        
    
    }
    
}

